### ISSUE TITLE
FireFly broke after assigning budget/categories

### ISSUE THREAD

**Issue by yverminck** [firefly-iii/firefly-iii#652]
Hello,

I upgraded recently to 4.4.3 and since my installation returned a lot of errors. Mostly related to budget edition.

The first errors I got was the following:
Go to transactions, select multiple transactions and click edit. I would then get an error "Whoops" but I don't remember the specifics.

Lastly, this afternoon, I was updating transactions in bulk ( I found a workaround by selecting a Destination Account and setting a budget/category. Then it all went bankrupt.

I can reach the login page and when I try to login, i get the following error:
[begin]
Whoops! An error occurred.
Unfortunately, this error was not recoverable. Firefly III broke. The error is:

file_put_contents(/var/www/html/firefly-iii/storage/framework/cache/08/ba/08ba0039f73344d8f349b266257beab32c1004ee): failed to open stream: No such file or directory

More information about this error may be available in the log files. Search for the error above and a stack trace will appear. This stack trace may help you track down what's going wrong.

If you cannot find the cause of this error or when you need some help, please open a ticket on Github.
[end]

I went to the specific place /cache/08 and effectively, the ba folder is missing.
I also went in the logs and i get the following:
[begin]
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:07] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #21  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:08] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[2017-05-03 15:29:09] production.ERROR: Cannot find tag #22  
[end]

Can you help me restore my installation. Obviously, I don't want to loose all my data.
Is there a way to extract all transactions without querying the db? Not that I cannot do it, but i would prefer something easier to re-import.

Thank you in advance,

Yoann

**Comment by yverminck** [comment]
I got it back by reverse-engineering the error and copying a file from another directory.
I hope nothing is broken

**Comment by JC5** [comment]
I'll check what is going on! Which file did you copy?

**Comment by JC5** [comment]
Alright let me know if you run into this issue again!

**Comment by yverminck** [comment]
Sorry, I was on holiday this weekend.
I will use Firefly this week and let you know.

About the file I used:
I created the folder "ba" in the folder /cache/08. Then I copied a file from another subfolder in the /cache folder. I renamed this file to "08ba0039f73344d8f349b266257beab32c1004ee".  I reloaded Firefly and complained about a property inside the file (\Models\TransactionCurrency). I did a search in the subfolders of /cache to find a file with that property inside and copied/renamed it. When I reloaded Firefly, it was okay again. I checked but it seems I didn't loose any data.
